strimzi-kafka-operator.yaml

```diff
@@ -0,0 +1,90 @@
+  version: 0.43.0
+  epoch: 0
+  description: Apache Kafka® running on Kubernetes
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - bash
+      - busybox
+      - ca-certificates-bundle
+      - coreutils
+      - curl
+      - docker # not gonna be needed but their script requires docker to be installed
+      - helm
+      - make
+      - maven
+      - openjdk-17
+      - openjdk-17-default-jvm
+      - unzip
+      - wget
+      - yq
+      - zip
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/strimzi/strimzi-kafka-operator
+      tag: ${{package.version}}
+      expected-commit: 23c0fb0726e1f212c6182eb251ff23697e72a353
+
+  - runs: |
+      export JAVA_HOME="/usr/lib/jvm/java-17-openjdk"
+      # needs shellcheck
+      scversion="stable"
+      wget -qO- "https://github.com/koalaman/shellcheck/releases/download/${scversion?}/shellcheck-${scversion?}.linux.${{build.arch}}.tar.xz" | tar -xJv
+      cp "shellcheck-${scversion}/shellcheck" /usr/bin/
+
+      for i in mockkube config-model certificate-manager operator-common; do
+        make -C $i MVN_ARGS="-B -DskipTests -Dmaven.javadoc.skip=true" java_build
+        make -C $i MVN_ARGS="-DskipTests -Dmaven.javadoc.skip=true -e -V -B" java_install
+      done
+
+      mvn clean
+      mvn install -B -DskipTests -Dmaven.javadoc.skip=true
+      make MVN_ARGS="-B -DskipTests -Dmaven.javadoc.skip=true" java_build
+      make MVN_ARGS="-DskipTests -Dmaven.javadoc.skip=true  -e -V -B" java_install
+
+      # Set the source and destination directories
+      SRC_DIR="./docker-images/artifacts/binaries"
+      DEST_DIR="opt/strimzi"
+      mkdir -p "${{targets.contextdir}}"/$DEST_DIR/bin
+      mkdir -p "${{targets.contextdir}}"/$DEST_DIR/lib
+      mv ./launch_java.sh ${{targets.contextdir}}/$DEST_DIR/bin/launch_java.sh
+
+
+      # List of file name patterns, space-separated
+      file_patterns="topic-operator user-operator cluster-operator kafka-init"
+
+      # Iterate over each pattern and match the corresponding zip files
+      for file in $file_patterns; do
+          # Find the zip file that matches the pattern
+          zipfile=$(ls "$SRC_DIR"/${file}-*-dist.zip 2>/dev/null)
+
+          # Check if the zip file exists
+          if [[ -f "$zipfile" ]]; then
+              # Extract the zip file into the destination directory
+              unzip -o "$zipfile" -d tmp
+
+              cp -r tmp/lib/* "${{targets.contextdir}}"/$DEST_DIR/lib/
+              cp -r tmp/bin/* "${{targets.contextdir}}"/$DEST_DIR/bin/
+
+              # Check if the unzip command was successful
+              if [[ $? -eq 0 ]]; then
+                  echo "Successfully unzipped $zipfile to $DEST_DIR"
+              else
+                  echo "Failed to unzip $zipfile"
+              fi
+          else
+              echo "No zip file found for pattern: ${file}-*-dist.zip"
+          fi
+      done
+
```

Comment:
Did we try adding some tests?

Reply:
I know some of these Java apps can be tricky to add melange test coverage for, but it might be possible

Reply:
I added some test but I'm not sure I could go further since the app is failing while detecting the k8s version, I noted it to the tests as a comment.

Reply:
I found similar issues in the past:

- https://github.com/strimzi/strimzi-kafka-operator/issues/6848
- https://github.com/strimzi/strimzi-kafka-operator/issues/2920

Reply:
Maybe we can just add a failing test that do something like: `grep Detection of Kuberetes version failed`?

Reply:
I removed the tests due to inconsistency between local and CI, I tested this package by creating an image locally, so, it should be bullet-proof 🔫